src/components/input/input.ts

```diff
@@ -0,0 +1,168 @@
+ */
+@customElement(`${prefix}-input`)
+export default class BXInput extends LitElement {
+  /**
+   * May be any of the standard HTML autocomplete options
+   */
+  @property()
+  autocomplete = '';
+
+  @property({ type: Boolean })
+  autofocus = false;
+
+  @property({ type: Boolean, reflect: true })
+  disabled = false;
+
+  /**
+   * The helper text. Corresponds to `helper-text` attribute.
+   */
+  @property({ attribute: 'helper-text' })
+  helperText = '';
+
+  @property()
+  id = '';
+
+  @property({ type: Boolean, reflect: true })
+  invalid = false;
+
+  /**
+   * The label text. Corresponds to `label-text` attribute.
+   */
+  @property({ attribute: 'label-text' })
+  labelText = '';
+
+  @property()
+  name = '';
+
+  @property()
+  pattern = '';
+
+  @property({ reflect: true })
+  placeholder = '';
+
+  @property({ type: Boolean, reflect: true })
+  readonly = false;
+
+  /**
+   * Boolean property to set the required status
+   */
+  @property({ type: Boolean, reflect: true })
+  required = false;
+
+  /**
+   * The type of the input. Can be one of the types listed in the INPUT_TYPE enum
+   */
+  @property({ reflect: true })
+  type = INPUT_TYPE.TEXT;
+
+  /**
+   * The validity message. Corresponds to `validity-message` attribute.
+   * If present and non-empty, this multi select shows the UI of its invalid state.
+   */
+  @property({ attribute: 'validity-message' })
+  validityMessage = '';
+
+  /**
+   * The value of the input.
+   */
+  @property({ reflect: true })
+  value = '';
+
+  /**
+   * Unique ID used for ID refs.
+   */
+  protected _uniqueId = Math.random()
+    .toString(36)
+    .slice(2);
+
+  protected get _inputId() {
+    return this.id || this._uniqueId;
+  }
+
+  render() {
+    const invalidIcon = WarningFilled16({ class: `${prefix}--text-input__invalid-icon` });
+
+    const inputClasses = classnames(`${prefix}--text-input`, {
+      [`${prefix}--text-input--invalid`]: this.invalid,
+    });
+
+    const labelClasses = classnames(`${prefix}--label`, {
+      [`${prefix}--label--disabled`]: this.disabled,
+    });
+
+    const helperTextClasses = classnames(`${prefix}--form__helper-text`, {
+      [`${prefix}--form__helper-text--disabled`]: this.disabled,
+    });
+
+    return html`
+      <label class="${labelClasses}" for="${this._uniqueId}">
+        <slot name="label-text">
+          ${this.labelText}
+        </slot>
+      </label>
+      <div class="${helperTextClasses}">
+        <slot name="helper-text">
+          ${this.helperText}
+        </slot>
+      </div>
+      <div class="${prefix}--text-input__field-wrapper" ?data-invalid="${this.invalid}">
+        ${this.invalid ? invalidIcon : null}
+        <input
+          ?autocomplete="${this.autocomplete}"
+          ?autofocus="${this.autofocus}"
+          class="${inputClasses}"
+          ?data-invalid="${this.invalid}"
+          ?disabled="${this.disabled}"
+          id="${this._inputId}"
+          invalid="${this.invalid}"
```

Comment:
Couldn't find `invalid` as a native property of `<input>`, so wondering if you have any intent behind `invalid` attribute here...?

Reply:
Huh, I could swear that was an attribute ... My bad!

Reply:
👍 No problem at all!